request.go

```diff
@@ -61,10 +62,38 @@ func (r *Request) Encode(v interface{}) {
 	r.Header.Set("Content-Type", "application/json")
 }
 
-// Decode de-serialises the JSON body into the passed object.
+// EncodeAsProtobuf serialises the passed object as protobuf into the body
+func (r *Request) EncodeAsProtobuf(m proto.Message) {
```

Comment:
Is this deliberately exported? I see the counterpart on `Response` is not exported, so curious why it's asymmetrical.